### ISSUE TITLE
Background Image Position

### ISSUE THREAD

**Issue by fernandosantucci** [telega1/fastdial#12]
How do I set the position of a background image on browser area resolution of Fast Dial page?

For example: Repeat, Zoom, Fill, Horizontal Fit, Vertical Fit or Center.

In have a huge wallpaper image of 2880x1620 pixels, but the resolution of my monitor is 1360x768 pixels. I know I could resize the image, but I connect my laptop in TVs and projectors with frequency and the image would be small for that devices.

May be you need create a new field "Image Position" in preferences.xul dialog box, appearance pane, page tab.

References:
https://developer.mozilla.org/en-US/docs/Web/CSS/background-position
https://developer.mozilla.org/en-US/docs/Web/CSS/background-repeat

**Comment by telega1** [comment]
By default, when you assign a background image in preferences, Fast Dial uses the following css settings:
        background-repeat   : repeat;
        background-position : top left;
        background-size     : auto;
Also there's a checkbox in preferences to the left of the background image textbox, which means you want to "center" your image. If you check it, Fast Dial uses the following settings:
        background-repeat   : no-repeat;
        background-position : center center;
        background-size     : cover;
Also, you can create a custom "style.css" file and zip it to make a custom theme with any other css settings. However in that case if user assigns other background image to your theme, these setting will get overridden.

**Comment by fernandosantucci** [comment]
Appeal to customize the file **style.css** is not a viable alternative for most users of Fast Dial, not programmers.

I'm with the latest version 4.15.1 updated and [Preferences -> Appearance pane -> Page tab] there is no checkbox next to the background **Image textbox**.

In [Preferences -> General pane], next to the **Size textbox**, there is a **Fixed checkbox** that applies the CSS attributes width and height of the thumbnail, not the image background.

I compared the version I'm running in production with the code of **preferences.xul** and **Preferences.js** of master on Github repository and found that there is a **Center checkbox** in the method below toggles the CSS attributes. However, you need provide this version on Mozilla Addons repository.

Still, I think it would be more flexible to more positioning options for the background image because the images of wallpaper are very large. Stay here my suggestion.

```
<label value="&image;:"/>
<hbox>
    <checkbox id="center" oncommand="saveStyle();"/>
    <textbox id="page" flex="1" onchange="setImage('page', this.value);"/>
</hbox>
```

```
function initPageStyle() {
    var doc = content.document;
    fastdial.Dom.get("page").value = images["page"] = getImageURL(doc.body);
    var repeat = fastdial.Dom.css(doc.body, "background-repeat");
    fastdial.Dom.get("center").checked = repeat != "repeat";
}
```

**Comment by telega1** [comment]
> I'm with the latest version 4.15.1 of Fast Dial updated with Firefox 47 and [Preferences -> Appearance pane -> Page tab] there is no checkbox next to the background Image textbox.

Please see a screenshot of [Preferences > Appearance pane > Page tab], checkbox is in red circle. The dialog is in Russian, I'm sorry.
![screenshot](https://cloud.githubusercontent.com/assets/5007502/16502676/42883e8a-3f18-11e6-9457-0fba75e63010.png)

**Comment by fernandosantucci** [comment]
No. As I said, the **Center checkbox** does not appear in the path indicated. Follow the screenshot:
![screenshot-2016-07-02t01-17-52](https://cloud.githubusercontent.com/assets/5784007/16538210/9992e906-3ff3-11e6-9566-6d36e2a4e460.png)

**Comment by telega1** [comment]
Please try version from here: http://www.userlogos.org/files/extensions/fast-dial/dev/fastdial-4.15.2-fx.xpi

**Comment by fernandosantucci** [comment]
Canonical Ubuntu 16.04 LTS
http://www.ubuntu.com/desktop

**Comment by telega1** [comment]
Thanks. Did you try the version I mentioned? Did the checkbox work for you?
